Removed `app.config.chains.getAll()` references and `/communities` call during app init

mzparacha opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Link to Issue
Closes: 
- https://github.com/hicommonwealth/commonwealth/issues/2617
- https://github.com/hicommonwealth/commonwealth/issues/8882
- https://github.com/hicommonwealth/commonwealth/issues/8753

## Description of Changes
- Removed `app.config.*` from the `app` object and all references to `app.config.chains.getAll()`
- Removed `/communities` call during app init flow, i.e the communities list is now fully decoupled from the app state, and now the app assumes that it should fetch any community info it needs. This is different from the previous state where any community info was guaranteed to exist after app init.
- Refactored and cleaned up the communities page with some behavior changes that referenced `app.config.chains.getAll()`
    - Added Virtuoso list, this will now work on infinite scroll instead of the older way of rendering every community
    - Updated some filters and related UI, now the chain base filter only accepts a single value, instead of multiple bases
- Updated `/status` API, to return user joined `communities` list with `is_starred` boolean, deprecating the `starredCommunities` response and state from the status route and frontend.
- Updated communities list API to support the `network` filter
- Updated communities list API to return extra fields i.e `profile_count`, and `thread_count` from the response
- Fixed communities list API breaking due to zod error when certain communities had nullish data for fields not expecting that.
- Moved the community suggestion algo to API, and cleaned up frontend usage of `app.config.chains.getAll()` that were used to calc suggested communities in the frontend
- Updated API's returning auth user addresses to include `Community: {id, base, ss58Prefix}` per address -- this was necessary for decoupling `app.config.chains` from `AddressInfo`
- Updated community update API to check for duplicate/already-taken custom domains and removed that check from the front end. Now when super admin is updating the custom domain of a community, we show duplicate/already-taken errors on form submit, rather than live as user types. This was needed as we don't store the whole chains/communities list on the front end anymore.
- When creating a community, the user will no longer receive 'live' duplicate/already-taken errors on the community name field, they will be shown when the user submits the form. This was needed as we don't store the whole chains/communities list on the front end anymore.
- Cleaned up unnecessary references to `AddressInfo` to reduce the load on `DISCOURAGED_NONREACTIVE_fetchProfilesByAddress` calls in `models/Account.ts`
- Updated `/status` route to return `communityWithRedirects: {id, redirect}[]`, this was necessary as we removed fetching all the communities during init and we needed a way to get community redirect IDs. The app init state is also refactored to utilize this logic.

## "How We Fixed It"
N/A

## Test Plan
- Test all auth methods, make sure there are no regressions.
- Test all communities flow, make sure there no regressions.
- Test updating custom domain from admin panel, make sure it works.
- Test communities page with all the different filters, make sure all filters work, and scrolling leading to more communities being loaded.
- Testing address linking works correctly with community admin and non-admin account (IMP: test with community admin/non-admin accounts, not with site admin)
- Test account creating, join community step, and make sure you get matching/proper community suggestions per user wallet
- Test community starring/unstarring works correctly
- Test edit profile page works correctly and you are able to unlink addresses properly

## Deployment Plan
N/A

## Other Considerations
- Paves the way to remove older (non-trpc based) `/communities` and `/communities?active=true` endpoints -- Ticketing need
- In the communities page, there are a bunch of calls to `https://base-mainnet.g.alchemy.com/v2/{id}`, and in most cases the id is same, this could possibly be cached in react query to reduce network load as a followup.

mzparacha commented:
Update: need to fix the issues mentioned in development plan -> then self test -> then ready for review

masvelio commented:
will this PR close [this ticket](https://github.com/hicommonwealth/commonwealth/issues/8405)?

mzparacha commented:
> will this PR close https://github.com/hicommonwealth/commonwealth/issues/8405? if yes feel free to attach it as well (or close it as duplicate). If no, please edit the description what else needs to be done in context of app.config.chains / chain store

This pr/ticket (https://github.com/hicommonwealth/commonwealth/issues/2617) will close #8405. Closing #8405 as duplicate. Thanks for the catch.

mzparacha commented:
Update: self-testing.

mzparacha commented:
I am not getting / or getting different CI issues in local env. Not sure what's happening here, will investigate.

mzparacha commented:
Tried testing buy/sell stake, but my metamask wallet is crashing for some reason (not sure if coz of the applied changes). Looking into it.

<img width="1710" alt="image" src="https://github.com/user-attachments/assets/eec1d906-e2ec-470d-ac88-95da35689a6c">

jnaviask commented on packages/commonwealth/client/scripts/state/ui/user/user.ts:
```diff
@@ -41,7 +47,7 @@ export const userStore = createStore<UserStoreProps>()(
     knockJWT: '',
     addresses: [],
     activeCommunity: null,
-    starredCommunities: [],
```
Are we eliminating starred communities in the global sidebar?

jnaviask commented on libs/model/src/community/GetCommunities.query.ts:
```diff
@@ -109,7 +117,7 @@ export function GetCommunities(): Query<typeof schemas.GetCommunities> {
                         }
                         ${
-                          filtering_tags
+                          filtering_tags && loose_filter !== 'tag_ids'
```
What is the purpose of `loose_filter` here?

mzparacha replied:
I couldn't think of a better name for this tbh, basically, its for relevance sort of matching tag_ids, results with matching tags ids will be displayed first, and then those who don't have any tag or non-matching tag ids

mzparacha replied:
I updated the name to `relevance_by` and added a comment on what would happen if its present.

masvelio commented on packages/commonwealth/client/scripts/views/components/SublayoutHeader/useUserMenuItems.tsx:
```diff
@@ -122,20 +123,52 @@ const useUserMenuItems = ({
+        // making a fresh query to get chain and community info for this address
+        // as all the necessary fields don't exist on user.address, these should come
+        // from api in the user address response, and the extra api call here removed
+        const community = await EXCEPTION_CASE_VANILLA_getCommunityById(
```
same, in hook, can't we use useGetCommunityByIdQuery?

mzparacha replied:
answered [here](https://github.com/hicommonwealth/commonwealth/pull/8877#discussion_r1723621623)

masvelio commented on packages/commonwealth/client/scripts/views/components/component_kit/new_designs/CWRelatedCommunityCard/CWRelatedCommunityCard.tsx:
```diff
@@ -59,7 +49,7 @@ export const CWRelatedCommunityCard = ({
 
   const { stakeEnabled, stakeValue, stakeChange } = useCommunityCardPrice({
-    communityId: community?.id,
+    community: community,
```
shorthand `community`

masvelio commented on packages/commonwealth/client/scripts/views/components/user/user_block.tsx:
```diff
@@ -1,152 +0,0 @@
-  hideAvatar?: boolean;
-};
-
-export const UserBlock = ({
```
we didnt use it anywhere?

mzparacha replied:
yup

masvelio commented on packages/commonwealth/client/scripts/views/modals/WelcomeOnboardModal/steps/JoinCommunityStep/JoinCommunityStep.tsx:
```diff
@@ -1,87 +1,91 @@
+    .map((t) => t.id || 0)
+    .filter((id) => id);
+
+  const { data: communitiesList, isLoading: isLoadingCommunities } =
```
amazing 💪

masvelio commented on packages/commonwealth/client/scripts/views/pages/AdminPanel/UpdateCustomDomainTask.tsx:
```diff
@@ -72,19 +73,6 @@ const UpdateCustomDomainTask = () => {
     }
-
-    // there's probably a better way to remove prefixes for duplicate finding purposes
-    const existingCustomDomain = app.config.chains
```
why we dont need that anymore?

mzparacha replied:
its moved to api
> Updated community update API to check for duplicate/already-taken custom domains and removed that check from the front end. Now when super admin is updating the custom domain of a community, we show duplicate/already-taken errors on form submit, rather than live as user types. This was needed as we don't store the whole chains/communities list on the front end anymore.

masvelio reviewed (commented):
went through the code, left some comments. Will try to click around tomorrow

mzparacha commented on packages/commonwealth/client/scripts/views/Layout.tsx:
```diff
@@ -89,17 +89,18 @@ const LayoutComponent = ({
           if (user.isLoggedIn) {
             await updateActiveCommunity({
               communityId: community?.id || '',
+            }).then(() => {
```
move the `.then` stuff in the outer scope

mzparacha commented on packages/commonwealth/client/scripts/controllers/app/login.ts:
```diff
@@ -203,7 +203,7 @@ export function updateActiveUser(data) {
       emailNotificationInterval: '',
       knockJWT: '',
       addresses: [],
-      starredCommunities: [],
```
The newer `communities` does a few more things like listing all joined-communities, including holding starred communities state. Plus its expected to expand in coming PR's for lazy loading / chains cleanup

masvelio reviewed (approved):
1. starring community does not work properly for me, when I star it, it is not added to the sidebar. When I refresh it is there. When I unstar it, it gets removed from the sidebar. 

https://github.com/user-attachments/assets/8fd2eb7a-65e6-40b8-ab42-ed5fcdc5e0a2

2. search throws 500 (I think I already mentioned it or something similar in the near past. If this is not your bug, lets try to ask Ilya to test it and file an issue)

![image](https://github.com/user-attachments/assets/3ce87a0b-9592-4c73-877c-b07df3cc0f1e)

3. saving community tags seems to work but when I refresh the tags are wiped out


https://github.com/user-attachments/assets/c430d7b5-c3d4-4f74-925e-2e37e411b2f2

I didnt find any other major bugs, thus I am accepting in advance. Great work, appreciate all those cleanups!

mzparacha commented:
Thanks for the review @masvelio. Addressed the issues

1. was a state issue, fixed in https://github.com/hicommonwealth/commonwealth/pull/8877/commits/daf4d9f34386f0afdccc0a0d62bf6ac391a202a0
2. kurtis fixed this in a different pr, I merged master just in case.
3. was a state issue, fixed in https://github.com/hicommonwealth/commonwealth/pull/8877/commits/c6e0e5771fb1b68b65cf3c4543539bd34bfdbf64

jnaviask reviewed (approved):
Approving because I only found small bugs for follow-up PRs, nothing preventing merging.

1. Joining a community doesn't add it to the "your communities" sidebar list until you refresh.
2. These trending communities seem wrong / should have some text rather than just "0"
![2024-08-22-111207_712x723_scrot](https://github.com/user-attachments/assets/514183e8-5e4e-4627-8292-057c8f67802c)
3. No member counts on any communities on explore page.
![2024-08-22-111244_956x711_scrot](https://github.com/user-attachments/assets/631f876e-f6a2-4dd7-b2b4-2df00316ef08)

mzparacha commented:
Thanks for the feedback @jnaviask. Please see

1. This is a bug indeed, but it was already in master. I will fix it here tho
2. There is a migration that updated communities table `thread_count` to `lifetime_thread_count`, not sure if you ran that?
3. same as 2.

jnaviask commented:
@mzparacha I don't see lifetime_thread_count anywhere in the repo, and no migrations run when I execute `pnpm migrate-db`

mzparacha commented:
Sry, i confused that with #8936, apologies. Fixing the issues

mzparacha commented:
@jnaviask fixed the issues, plz try when you get a chance.

jnaviask commented:
(1) is fixed but 2/3 still seem outstanding

mzparacha commented:
yes that changes a few things regarding counts. BTW I am still not able to replicate 3 in my machine
